src/views/activity/components/transaction-card/transaction-card.view.tsx

```diff
@@ -1,45 +1,34 @@
 import { FC } from "react";
+import { formatEther } from "ethers/lib/utils";
+import { useNavigate } from "react-router-dom";
 
 import useTransactionCardStyles from "src/views/activity/components/transaction-card/transaction-card.styles";
 import { ReactComponent as TransferL1Icon } from "src/assets/icons/l1-transfer.svg";
 import { ReactComponent as TransferL2Icon } from "src/assets/icons/l2-transfer.svg";
 import { ReactComponent as ReloadIcon } from "src/assets/icons/spinner.svg";
 import Typography from "src/views/shared/typography/typography.view";
 import Card from "src/views/shared/card/card.view";
-import { getTimeFromNow } from "src/utils/time";
-import { useNavigate } from "react-router-dom";
 import routes from "src/routes";
 import Icon from "src/views/shared/icon/icon.view";
-import { TransactionStatus, getTransactionStatusText } from "src/domain";
-import { useEnvContext } from "src/contexts/env.context";
+import { Transaction, getTransactionStatusText } from "src/domain";
 
 export interface TransactionCardProps {
-  target: "l1" | "l2";
-  id: number;
-  timestamp: number;
-  token: "eth" | "dai";
-  status: TransactionStatus;
-  amount: number;
+  transaction: Transaction;
+  onClaim: () => void;
 }
 
-const layerIcons = {
-  l1: TransferL1Icon,
-  l2: TransferL2Icon,
-};
-
-const TransactionCard: FC<TransactionCardProps> = ({
-  id,
-  target,
-  timestamp,
-  token,
-  amount,
-  status,
-}) => {
+const TransactionCard: FC<TransactionCardProps> = ({ transaction, onClaim }) => {
+  const { status, destinationNetwork, depositCount, amount } = transaction;
   const classes = useTransactionCardStyles();
   const navigate = useNavigate();
-  const env = useEnvContext();
-  const LayerIcon = status !== "completed" && status !== "failed" ? ReloadIcon : layerIcons[target];
-  const actionText = target === "l1" ? "Transfer to L1" : "Transfer to L2";
+  const LayerIcon =
+    status !== "completed"
+      ? ReloadIcon
+      : destinationNetwork.key === "ethereum"
+      ? TransferL1Icon
+      : TransferL2Icon;
+  const actionText = destinationNetwork.key === "ethereum" ? "Transfer to L1" : "Transfer to L2";
```

Comment:
I would render both `LayerIcon` and `actionText` directly in the `JSX` as they are only used once.

Reply:
Done